resolwe_bio/tools/goea.py

```diff
@@ -33,27 +34,36 @@ def main():
     org_features = res.feature.filter(source=args.source_db, species=args.species, feature_id=genes)
 
     if len(org_features) == 0:
-        print('{"proc.error":"No genes were fetched from the knowledge base."}')
+        print(error("No genes were fetched from the knowledge base."))
         exit(1)
 
     if args.source_db == args.target_db:
         target_ids = genes
     else:
-        features = res.mapping.filter(
+        mapping_res = res.mapping.filter(
             source_db=args.source_db,
             source_species=args.species,
             target_db=args.target_db,
             target_species=args.species,
-            source_id=genes)
+            source_id=genes
+        )
 
-        if len(features) == 0:
-            print('{"proc.error":"Failed to map features."}')
+        if len(mapping_res) == 0:
+            print(error("Failed to map features."))
             exit(1)
 
-        target_ids = [str(feature.target_id) for feature in features]
+        mappings = {}
+        for m in mapping_res:
+            if m.source_id in genes:
+                if m.source_id not in mappings:
+                    mappings[m.source_id] = m.target_id
+                else:
+                    print(warning("Mapping {} returned multiple times.".format(m)))
```

Comment:
Add a single quotes around `{}`.

Reply:
Do you think it is necessary?

Here is an example output of the current code:
```
Mapping <Mapping source_db='NCBI' source_id='56535' target_db='MGI' target_id='MGI:1929646'> returned multiple times.
```

Reply:
Ok, leave it like this.